src/modules/shopPage/components/FilterMenu/FilterFullMenu/FilterFullMenuContainer.tsx

```diff
@@ -0,0 +1,285 @@
+
+import Tabs from "@mui/material/Tabs";
+import Tab from "@mui/material/Tab";
+import Typography from "@mui/material/Typography";
+import Box from "@mui/material/Box";
+import { IconButton, styled } from "@mui/material";
+import CloseIcon from "@mui/icons-material/Close";
+
+import FilterFullMenuInput from "./FilterFullMenuInput";
+import FilterFullMenuWidthData from "./FilterFullMenuWidthData";
+import FilterFullMenuProfileData from "./FilterFullMenuProfileData";
+import FilterFullMenuDiametrData from "./FilterFullMenuDiametrData";
+import FilterFullMenuPriceData from "./FilterFullMenuPriceData";
+import FilterFullMenuSeasonData from "./FilterFullMenuSeasonData";
+import FilterFullMenuBrandData from "./FilterFullMenuBrandData";
+
+import WidthIcon from "../../../../../shared/components/Icons/WidthIcon";
+import ProfileIcon from "../../../../../shared/components/Icons/ProfileIcon";
+import SeasonIcon from "../../../../../shared/components/Icons/SeasonIcon";
+import BrandIcon from "../../../../../shared/components/Icons/BrandIcon";
+import PriceIcon from "../../../../../shared/components/Icons/PriceIcon";
+import DiametrIcon from "../../../../../shared/components/Icons/DiametrIcon";
+
+import FilterFullMenuData from "../../../../../config/FilterFullMenuData.json";
+import { FONTS, BASE_COLORS } from "../../../../../shared/constants";
+
+type TabPanelProps = {
+  children?: React.ReactNode;
+  index: number;
+  value: number;
+};
+
+type FilterData = {
+  headerTitle?: string;
+  asideHeader?: string;
+  text1?: string;
+  text2?: string;
+  text3?: string;
+  parametr?: string;
+  textForParametr?: string;
+};
+
+type FilterFullMenuDataType = {
+  Width?: FilterData;
+  Profile?: FilterData;
+  Diametr?: FilterData;
+  Price?: FilterData;
+  Season?: FilterData;
+  Brand?: FilterData;
+};
+
+type MenuData = {
+  inputComponent?: React.ElementType;
+  dataComponent: React.ElementType;
+} & FilterData;
+
+const StyledText = styled(Typography)({
+  fontSize: "14px",
+  marginTop: "18px",
+  fontFamily: FONTS.MAIN_TEXT_FAMILY,
+});
+
+function TabPanel(props: TabPanelProps) {
+  const { children, value, index, ...other } = props;
+
+  return (
+    <div
+      role="tabpanel"
+      hidden={value !== index}
+      id={`vertical-tabpanel-${index}`}
+      aria-labelledby={`vertical-tab-${index}`}
+      {...other}
+    >
+      {value === index && (
+        <Box sx={{ p: 0 }}>
+          <Box>{children}</Box>
+        </Box>
+      )}
+    </div>
+  );
+}
+
+function a11yProps(index: number) {
+  return {
+    id: `vertical-tab-${index}`,
+    "aria-controls": `vertical-tabpanel-${index}`,
+  };
+}
+
+function FilterFullMenuContainer() {
+  const dispatch = useDispatch();
+  const activeTabIndex = useSelector(selectActiveTabIndex);
+  const [value, setValue] = React.useState(activeTabIndex);
+  const [textData, setTextData] = useState<FilterFullMenuDataType>({});
+
+  function handleChange(event: React.SyntheticEvent, newValue: number) {
```

Comment:
use an arrow function and `useCallback` hook for all callbacks

Reply:
Done in all components